pulsar-client/src/main/java/org/apache/pulsar/client/impl/transaction/TransactionImpl.java

```diff
@@ -133,15 +133,9 @@ public CompletableFuture<Void> registerAckedTopic(String topic, String subscript
         if (checkIfOpen(completableFuture)) {
             synchronized (TransactionImpl.this) {
                 // we need to issue the request to TC to register the acked topic
-                return registerSubscriptionMap.compute(Pair.of(topic, subscription), (key, future) -> {
-                    if (future != null) {
-                        return future.thenCompose(ignored -> CompletableFuture.completedFuture(null));
-                    } else {
-                        return tcClient.addSubscriptionToTxnAsync(
-                                new TxnID(txnIdMostBits, txnIdLeastBits), topic, subscription)
-                                .thenCompose(ignored -> CompletableFuture.completedFuture(null));
-                    }
-                });
+                return registerSubscriptionMap.computeIfAbsent(Pair.of(topic, subscription), key ->
+                        tcClient.addSubscriptionToTxnAsync(
+                                new TxnID(txnIdMostBits, txnIdLeastBits), topic, subscription));
             }
         }
         return completableFuture;
```

Comment:
No need to `checkIfOpen`, because it will always be open.

```suggestion
    public CompletableFuture<Void> registerAckedTopic(String topic, String subscription) {
        CompletableFuture<Void> completableFuture = new CompletableFuture<>();
        synchronized (TransactionImpl.this) {
            // we need to issue the request to TC to register the acked topic
            return registerSubscriptionMap.computeIfAbsent(Pair.of(topic, subscription), key ->
                    tcClient.addSubscriptionToTxnAsync(new TxnID(txnIdMostBits, txnIdLeastBits), topic, subscription));
        }
    }
```